File: algorithm-puzzles/q43_2.js
N = 12345;

function trit(n) {
	if (n == 0) return 1;
	let a = 0;
	while (Math.pow(3, a + 1) <= n) {
		a++;
	}
	return Math.pow(2, a) + trit(Math.min(n - Math.pow(3, a), Math.pow(3, a) - 1));
}

console.log(trit(N));

File: algorithm-puzzles/q43_2.py
N = 12345


def trit(n):
    if (n == 0):
        return 1;
    a = 0
    while 3 ** (a + 1) <= n:
        a += 1
    return 2 ** 2 + trit(min([n - 3 ** a, 3 ** a - 1]))


print(trit(1));
